onmt/opts.py

```diff
@@ -107,6 +107,9 @@ def model_opts(parser):
                        choices=['dot', 'general', 'mlp'],
                        help="""The attention type to use:
                        dotprod or general (Luong) or MLP (Bahdanau)""")
+    group.add_argument('-average_attn', action='store_true',
```

Comment:
There could be other types of self attention layer in the future. So maybe it's best to directly accept a string value (cf. `self_attn_type` in `TransformerDecoderLayer`)